lib/screens/user_progress.dart

```diff
@@ -36,27 +36,47 @@ class UserProgress extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    BuiltMap<String, GameStatus> gameStatuses;
+    
+
+ StateContainer.of(context).setGameStatuses(BuiltMap<String, GameStatus>({
```

Comment:
Why are you setting the gameStatus in userprogress? You should only be reading the values